models/project_board.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+type (
+	// ProjectBoardType is used to represent a project board type
+	ProjectBoardType uint8
+
+	// ProjectBoardList is a list of all project boards in a repository
+	ProjectBoardList []*ProjectBoard
+)
+
+const (
+	// None is a project board type that has no predefined columns
+	None ProjectBoardType = iota
+
+	// BasicKanban is a project board type that has basic predefined columns
+	BasicKanban
+
+	// BugTriage is a project board type that has predefined columns suited to hunting down bugs
+	BugTriage
+)
+
+// ProjectBoard is used to represent boards on a project
+type ProjectBoard struct {
+	ID      int64 `xorm:"pk autoincr"`
+	Title   string
+	Default bool //if true it collects issues witch are not signed to a specific board jet
+
+	ProjectID int64 `xorm:"INDEX NOT NULL"`
+	RepoID    int64 `xorm:"INDEX NOT NULL"`
+	CreatorID int64 `xorm:"NOT NULL"`
+
+	CreatedUnix timeutil.TimeStamp `xorm:"INDEX created"`
+	UpdatedUnix timeutil.TimeStamp `xorm:"INDEX updated"`
+
+	Issues []*Issue `xorm:"-"`
+}
+
+// IsProjectBoardTypeValid checks if the project board type is valid
+func IsProjectBoardTypeValid(p ProjectBoardType) bool {
+	switch p {
+	case None, BasicKanban, BugTriage:
+		return true
+	default:
+		return false
+	}
+}
+
+func createBoardsForProjectsType(sess *xorm.Session, project *Project) error {
+
+	var items []string
+
+	switch project.BoardType {
+
+	case BugTriage:
+		items = setting.Repository.ProjectBoardBugTriageType
+
+	case BasicKanban:
+		items = setting.Repository.ProjectBoardBasicKanbanType
+
+	case None:
+		fallthrough
+	default:
+		return nil
+	}
+
+	if len(items) == 0 {
+		return nil
+	}
+
+	var boards = make([]ProjectBoard, 0, len(items))
+
+	for _, v := range items {
+		boards = append(boards, ProjectBoard{
+			CreatedUnix: timeutil.TimeStampNow(),
+			UpdatedUnix: timeutil.TimeStampNow(),
+			CreatorID:   project.CreatorID,
+			Title:       v,
+			ProjectID:   project.ID,
+			RepoID:      project.RepoID,
+		})
+	}
+
+	_, err := sess.Insert(boards)
+	return err
+}
+
+// NewProjectBoard adds a new project board to a given project
+func NewProjectBoard(board *ProjectBoard) error {
+	_, err := x.Insert(board)
+	return err
+}
+
+// DeleteProjectBoardByID removes all issues references to the project board.
+func DeleteProjectBoardByID(repoID, projectID, boardID int64) error {
```

Comment:
This shouldn't need to take a repoID, projectIDs should be unique within a Gitea instance.